rott/dukemusc.c

```diff
@@ -265,65 +266,33 @@ int MUSIC_StopSong(void)
 
     music_songdata = NULL;
     music_musicchunk = NULL;
+	music_songdatasize = 0;
     return(MUSIC_Ok);
 } // MUSIC_StopSong
 
-
-int MUSIC_PlaySong(unsigned char *song, int loopflag)
-{
-    //SDL_RWops *rw;
-
-    MUSIC_StopSong();
-
-    music_songdata = song;
-
-    // !!! FIXME: This could be a problem...SDL/SDL_mixer wants a RWops, which
-    // !!! FIXME:  is an i/o abstraction. Since we already have the MIDI data
-    // !!! FIXME:  in memory, we fake it with a memory-based RWops. None of
-    // !!! FIXME:  this is a problem, except the RWops wants to know how big
-    // !!! FIXME:  its memory block is (so it can do things like seek on an
-    // !!! FIXME:  offset from the end of the block), and since we don't have
-    // !!! FIXME:  this information, we have to give it SOMETHING.
-
-    /* !!! ARGH! There's no LoadMUS_RW  ?!
-    rw = SDL_RWFromMem((void *) song, (10 * 1024) * 1024);  // yikes.
-    music_musicchunk = Mix_LoadMUS_RW(rw);
-    Mix_PlayMusic(music_musicchunk, (loopflag == MUSIC_PlayOnce) ? 0 : -1);
-    */
-
-musdebug("Need to use PlaySongROTT.  :(");
-
-    return(MUSIC_Ok);
-} // MUSIC_PlaySong
-
-// ROTT Special - SBF
-int MUSIC_PlaySongROTT(unsigned char *song, int size, int loopflag)
+int MUSIC_PlaySong(unsigned char *song, int size, int loopflag)
 {
-    char *filename;
-    int handle;
-    
     MUSIC_StopSong();
 
-    // save the file somewhere, so SDL_mixer can load it
-    filename = M_StringJoin(ApogeePath, PATH_SEP_STR, "tmpsong.mid", NULL);
-    handle = SafeOpenWrite(filename);
-    
-    SafeWrite(handle, song, size);
-    close(handle);
-    
-    music_songdata = song;
+	// create rw
+	SDL_RWops *rw = SDL_RWFromConstMem(song, size);
+	if (rw == NULL) {
+		return MUSIC_Error;
+	}
 
-    // finally, we can load it with SDL_mixer
-    music_musicchunk = Mix_LoadMUS(filename);
-    free(filename);
+    // load with SDL_mixer
+    music_musicchunk = Mix_LoadMUS_RW(rw, SDL_TRUE);
```

Comment:
The `SDL_TRUE` in the second parameter means that we don't have to care about freeing the `rw` stuff ourselves, right?

Reply:
Correct